packages/core/src/lib/code-snippet.svelte

```diff
@@ -0,0 +1,227 @@
+<!--
+@component
+  
+For rendering readonly blocks of code. Uses `prism` for syntax highlighting:
+
+https://prismjs.com
+
+```svelte
+<CodeSnippet language="json" code="{ myThing: true }">
+    This text captions the code snippet.
+</CodeSnippet>
+```
+-->
+<svelte:options immutable />
+
+<script
+  context="module"
+  lang="ts"
+>
+import { highlightElement, plugins } from 'prismjs';
+import PrismPackage from 'prismjs/package.json';
+
+export type CodeSnippetTheme = 'vs' | 'vsc-dark-plus';
+
+const PRISM_VERSION = PrismPackage.version;
+</script>
+
+<script lang="ts">
+import { createEventDispatcher, onMount } from 'svelte';
+import cx from 'classnames';
+
+import type { IconName } from '$lib';
+import IconButton from './button/icon-button.svelte';
+
+/**
+ * The language to use for syntax highlighting. Must be a language supported by
+ * prism: https://prismjs.com/#supported-languages
+ */
+export let language: string;
+
+/**
+ * The code to render in the code-snippet.
+ */
+export let code: string;
+
+/**
+ * The theme to use for syntax highlighting. Must be a theme provided by
+ * prism-themes: https://github.com/PrismJS/prism-themes#available-themes
+ *
+ * Currently only the `vs` and `vsc-dark-plus` themes are supported.
+ */
+export let theme: CodeSnippetTheme = 'vs';
+
+/**
+ * Whether or not to show a copy-to-clipboard button.
+ */
+export let showCopyButton = true;
+
+/**
+ * Some prism languages have dependencies. For example, C++ requires C. If the
+ * passed `language` has dependencies, they must be included here to be loaded.
+ *
+ * See: https://prismjs.com/plugins/autoloader/
+ */
+export let dependencies: string[] = [];
+
+/**
+ * We use the prism autoloader to handle loading in language grammar files. The
+ * default path for the grammar files is a CDN link so we don't have to include
+ * grammar files in our bundle. If you prefer to point to your own grammars and
+ * bypass the CDN, define the path to those files.
+ *
+ * See: https://prismjs.com/plugins/autoloader/
+ */
+export let grammarsPath = `https://cdnjs.cloudflare.com/ajax/libs/prism/${PRISM_VERSION}/components/`;
```

Comment:
After the conversation around whether to use the CDN or not, I added this as a potential short-circuit. This will allow users to define their own path to grammar files, if they don't want to lazy-load them from the CDN and grab them from a local static path for example. This may not be necessary, but I could see us including the grammars for SDK languages we support so we can fetch those locally in `app`.